src/finiteschemesolve.jl

```diff
@@ -5,8 +5,8 @@
 ##############################################################################
 
 # Implicit time step
-function implicit_timestep(F!, ypost, Δ; is_algebraic = fill(false, size(ypost)...), iterations = 100, verbose = true, method = :newton, autodiff = :forward, maxdist = sqrt(eps()), J0c = (nothing, nothing))
-    F_helper!(ydot, y) = (F!(ydot, y) ; ydot .+= .!is_algebraic .* (ypost .- y) ./ Δ)
+function implicit_timestep(F!, ypost, Δ; is_algebraic = fill(false, size(ypost)...), iterations = 100, verbose = true, method = :newton, autodiff = :forward, maxdist = sqrt(eps()), J0c = (nothing, nothing), y̲ = fill(-Inf, length(ypost)), ȳ = fill(Inf, length(ypost)), reformulation = :smooth)
+    F_helper!(ydot, y) = (F!(ydot, y) ; ydot .+= .!is_algebraic .* (ypost .- y) ./ Δ; ydot .*= -1) # follow the sign convention by mcpsolve
```

Comment:
I don't really understand this line. Why are you adding another keyword argument? Also it seems like you're changing F_helper for any solution, not just the one using mcpsolve

Reply:
It is due to the sign convention in mcpsolve. I added some comments here in the new commit:

The HJBVI is `min{-ydot, y - y̲}=0`, so when `y==y̲, ydot ≤ 0` (remember here `ydot = u + ∂y/∂x dx - ρy + ẏ`), while the sign convention in mcpsolve is when` y==y̲, ydot>=0`.

and yes I change it for any solution but usual HJBs are not affected since they require `ydot==0` anyway.